docs/our-approach-to-data.md

```diff
@@ -0,0 +1,267 @@
+
+1. `mapStateToProps`: A function which, given the store state, returns props to be passed to the connected component. This is used to satisfy the need to make data available to the component.
+2. `mapDispatchToProps`: A function which, given the store dispatch method, returns props to be passed to the connected component. This is used to satisfy the need to allow the component to update the store state.
+
+As an example, consider a component which renders a Delete button for a given post. We want to display the post title as a label in the delete button, and allow the component to trigger the post deletion when clicked.
+
+```jsx
+function PostDeleteButton( { label, delete } ) {
+	return (
+		<button onClick={ delete }>
+			{ this.translate( 'Delete "%s"', { args: [ label ] } ) }
+		</button>
+	);
+}
+
+export default connect(
+	( state, ownProps ) => {
+		return {
+			label: getSitePost( ownProps.siteId, ownProps.postId ).title
+		};
+	},
+	( dispatch, ownProps ) => {
+		return {
+			delete: () => dispatch( deleteSitePost( ownProps.siteId, ownProps.postId ) )
+		};
+	}
+)( PostDeleteButton );
+```
+
+At this point, you might observe that the visual elements rendered in `<PostDeleteButton />` aren't very specific to posts and could probably be reused in different contexts. This is a good observation to make, and in this case it might make sense to split the visual component to its own separate file (e.g. `client/components/delete-button/index.jsx`). You should try to identify these opportunities as often as possible. Since the `connect` wrapping function is detached from the component declaration in the file above, it should not be difficult to separate the two.
+
+#### Fetching components 
+
+Fetching components accept as few props as possible to describe the data needs of the context in which they're used. They are responsible for dispatching the actions that fetch the desired data from the WordPress.com REST API. They should neither accept nor render any children.
+
+The benefits of fetching components are that they (a) are reusable, (b) take advantage of React's lifecycle methods to ensure that data needs are kept in sync, and (c) can be used by [connected "App Components"](https://wpcalypso.wordpress.com/devdocs/app-components) to maintain their self-sufficiency. That they neither accept nor render children eliminates the need for ancestor components to concern themselves with the data needs of leaf components and [can be more performant](https://www.youtube.com/watch?v=KYzlpRvWZ6c&t=1137).
+
+When creating a component that needs to consume data, we can simply include a fetching component as a child of that component, consuming from the state selectors populated by the results of the network request initiated by the fetching component.
```

Comment:
>maybe `include a fetching component as a child or sibling of that component`?

The idea would be that any connected app component would be responsible for rendering its own fetching component such to enable its "droppability".

It's a little more unclear how to use fetching components in the context of a page: should any connected component render a fetching component if it needs data? I'd argue from a pragmatic standpoint, it probably makes more sense to render the fetching components once toward the top of the page's render hierarchy, unless we know that specific descendent components are to be reused between sections. These could be classified as siblings of the components from which the data is consumed.

I'll try to make this a bit more clear here.